### DISCUSSION TITLE
Script variable overflow causes exception (I think)

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#15769] [Q&A] [answered]
I was working on my BMS script and was adding chars one by one to a string variable in the >S section, but forgot the reset the variable after each loop, so it very quickly filled up and crashed the device, which then restarted and started filling up the variable again. (I'm pretty sure that's what caused my crashes, since when I clear the variable each loop, the problem goes away)

I had enough time between crashes to disable the script and fix the problem, but it was basically in a boot loop.

Are there any existing checks for variable overflows or boot loop protection for scripting that we could implement for users like me who do silly things?

**Comment by gemu2015** [comment, marked as answer]
There is no extensive error checking implemented neither for syntax errors nor variable overflows
This is because we want to keep flash space low.
Boot loop protection is implemented in tasmota and stops the script the same way as rules since it uses the same enable bit
If you post the script snipped that causes the overflow I can take a look how to prevent that

**Comment by pkkrusty** [comment]
I’ll see if I can find it/recreate it. I was adding 4 or 12 chars per second to a string variable. Seemed like I had about 2-3 seconds to catch it before it crashed. Maybe it was just outside the window when the boot loop flag would get set.
